infra/build/developer-tools/build/scripts/module-swapper/cmd/swap.go

```diff
@@ -0,0 +1,212 @@
+func disableModules(f []byte, p string) ([]byte, error) {
+	if err := validatePaths([]string{p}); err != nil {
+		return nil, err
+	}
+	absPath, err := filepath.Abs(filepath.Dir(p))
+	if err != nil {
+		return nil, fmt.Errorf("Error finding example absolute path: %v", err)
+	}
+	strFile := string(f)
+	lines := strings.Split(strFile, linebreak)
+	for _, localModule := range localModules {
+		// check if current file has module/submodules references that should be swapped
+		if !strings.Contains(strFile, localModule.ModuleFQN) {
+			continue
+		}
+		// get relative path from example to local module
+		newModulePath, err := filepath.Rel(absPath, localModule.Dir)
+		if err != nil {
+			return nil, fmt.Errorf("Error finding relative path: %v", err)
+		}
+		for i, line := range lines {
+			if strings.Contains(line, fmt.Sprintf("\"%s\"", localModule.ModuleFQN)) && !strings.Contains(line, restoreMarker) {
+				// swap with local module and add restore point
+				lines[i] = strings.ReplaceAll(line, localModule.ModuleFQN, newModulePath) + fmt.Sprintf("# %s %s", restoreMarker, line)
+				// if next line is a version declaration, disable that as well
+				if i < len(lines)-1 && strings.Contains(lines[i+1], "version") {
+					lines[i+1] = fmt.Sprintf("# %s %s", restoreMarker, lines[i+1])
+				}
+			}
+		}
+	}
+	newExample := strings.Join(lines, linebreak)
+	// check if any swaps have been made
+	if newExample == strFile {
+		return nil, nil
+	}
+	// print diff info
+	log.Printf("Modifications made to file %s", p)
+	diff := difflib.UnifiedDiff{
+		A:        difflib.SplitLines(strFile),
+		B:        difflib.SplitLines(newExample),
+		FromFile: "Original",
+		ToFile:   "Modified",
+		Context:  3,
+	}
+	diffInfo, _ := difflib.GetUnifiedDiffString(diff)
+	log.Println(diffInfo)
+	return []byte(newExample), nil
+
+}
+
+// rootModuleName returns root module name from env var or path
+func rootModuleName(p string) string {
+	if os.Getenv("MODULE_NAME") != "" {
+		moduleName := strings.ReplaceAll(os.Getenv("MODULE_NAME"), "terraform-google-", "")
+		log.Printf("Module name set from envvar to %s", moduleName)
+		return moduleName
+	}
+	moduleName := strings.ReplaceAll(filepath.Base(p), "terraform-google-", "")
+	log.Printf("Module name set from path to %s", moduleName)
+	return moduleName
+}
+
+// getTFFiles returns a slice of valid TF file paths
+func getTFFiles(path string) []string {
+	// validate path
+	if err := validatePaths([]string{path}); err != nil {
+		log.Fatal(err)
+	}
+	var files = make([]string, 0)
+	filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+		if err != nil && info.IsDir() {
+			return nil
+		}
+		isTFFile, _ := filepath.Match(terraformExtension, filepath.Base(path))
+		if isTFFile {
+			files = append(files, path)
+		}
+		return nil
+	})
+	return files
+
+}
+
+func SwapModules(rootPath, moduleRegistryPrefix, moduleRegistrySuffix, subModulesDir, examplesDir string, restore bool) {
+	moduleName := rootModuleName(rootPath)
```

Comment:
Please add some lines between different sections of this function.

- moduleName
- blank
- localModules + subModulesPath + submods
- blank
- examplesPath + exampleTfFiles + processing

Reply:
done